rand_distr/tests/cdf.rs

```diff
@@ -195,3 +565,433 @@ fn binomial() {
         });
     }
 }
+
+#[test]
+fn geometric() {
+    fn cdf(k: i64, p: f64) -> f64 {
+        if k < 0 {
+            0.0
+        } else {
+            1.0 - (1.0 - p).powi(1 + k as i32)
+        }
+    }
+
+    let parameters = [0.3, 0.5, 0.7, 0.0000001, 0.9999];
+
+    for (seed, p) in parameters.into_iter().enumerate() {
+        let dist = rand_distr::Geometric::new(p).unwrap();
+        test_discrete(seed as u64, dist, |k| cdf(k, p));
+    }
+}
+
+#[test]
+fn hypergeometric() {
+    fn cdf(x: i64, n: u64, k: u64, n_: u64) -> f64 {
+        let min = if n_ + k > n { n_ + k - n } else { 0 };
+        let max = k.min(n_);
+        if x < min as i64 {
+            return 0.0;
+        } else if x >= max as i64 {
+            return 1.0;
+        }
+
+        (min..x as u64 + 1).fold(0.0, |acc, k_| {
+            acc + (ln_binomial(k, k_) + ln_binomial(n - k, n_ - k_) - ln_binomial(n, n_)).exp()
+        })
+    }
+
+    let parameters = [
+        (15, 13, 10),
+        (25, 15, 5),
+        (60, 10, 7),
+        (70, 20, 50),
+        (100, 50, 10),
+        // (100, 50, 49), // Fail case
+    ];
+
+    for (seed, (n, k, n_)) in parameters.into_iter().enumerate() {
+        let dist = rand_distr::Hypergeometric::new(n, k, n_).unwrap();
+        test_discrete(seed as u64, dist, |x| cdf(x, n, k, n_));
+    }
+}
+
+#[test]
+fn poisson() {
+    fn cdf(k: i64, lambda: f64) -> f64 {
+        if k < 0 || lambda <= 0.0 {
+            return 0.0;
+        }
+
+        1.0 - gamma_lr(k as f64 + 1.0, lambda)
+    }
+    let parameters = [
```

Comment:
Can you make this a test over `f64`? Then `1e9` should also pass.

We have to think about how to test over different floating point, but I would do everything in `f64` for now.

Reply:
I didn't use `f64` here because its `sample` method has both `f64` and `u64` implementations, and version 1.61.0 cannot specify which implementation to use. This causes the MSRV test to fail during compilation.

However, if I move the KS test to `benches`, it will only be tested in nightly, allowing it to be performed on `f64`.

Reply:
You can change the definition of `test_discrete` to replace the impl Trait with a generic